DropboxLike.Domain/Repositories/FileRepository.cs

```diff
@@ -47,127 +47,167 @@ public async Task<S3Response> UploadFileAsync(IFormFile file)
+    catch (AmazonS3Exception exception)
     {
-      _response.StatusCode = (int)ex.StatusCode;
-      _response.Message = ex.Message;
+      var message = $"{exception.StatusCode}: {exception.Message}";
+      return OperationResult<object>.Fail(exception, message, exception.StatusCode);
     }
-    catch (Exception ex)
+    catch (Exception exception)
     {
-      _response.StatusCode = 500;
-      _response.Message = ex.Message;
+      var message = $"{HttpStatusCode.InternalServerError}: {exception.Message}";
+      return OperationResult<object>.Fail(exception, message);
     }
-    return _response;
   }
 
-  public async Task<FileStreamResult> DownloadFileAsync(string fileId)
+  public async Task<OperationResult<File>> DownloadFileAsync(string fileId)
   {
-
-    // var results = await _applicationDbContext.FileModels
-    //     .FirstOrDefaultAsync(x => x.FileKey == fileId);
-
-    //var destinationFolderPath = $"/home/godfreyowidi/Downloads/TestsDowloads";
-
     try
     {
       var file = await _applicationDbContext.FileModels.FindAsync(fileId);
-
       if (file == null)
       {
         throw new FileNotFoundException("File not Found");
       }
 
-      var objectKey = file?.FileKey?.ToString();
-
-      ListObjectsV2Request listRequest = new ListObjectsV2Request
+      var listRequest = new ListObjectsV2Request
       {
         BucketName = _bucketName,
       };
-      ListObjectsV2Response listResponse = await _awsS3Client.ListObjectsV2Async(listRequest);
+      
+      // This code just ensure we retrieve only object names - not all metadata
+      // Caching of the retrieved names for 20 mins - server side caching.
+      var cache = new MemoryCache(new MemoryCacheOptions());
+      if (cache.TryGetValue<List<string>>(_bucketName, out var fileNames))
+      {
+        var fileCache = new File
+        {
+          ObjectNames = fileNames
+        };
+        return OperationResult<File>.Success(fileCache);
+      }
+      fileNames = new List<string>();
+      
+      ListObjectsV2Response listResponse;
+      do
+      {
+        listResponse =  await _awsS3Client.ListObjectsV2Async(listRequest);
+        
+        fileNames.AddRange(listResponse.S3Objects.Select(obj => obj.Key));
+
+        listRequest.ContinuationToken = listResponse.NextContinuationToken;
+      } while (listResponse.IsTruncated);
+
+      var cacheOptions = new MemoryCacheEntryOptions().SetSlidingExpiration(TimeSpan.FromMinutes(10));
+
+      cache.Set(_bucketName, fileNames, cacheOptions);
 
-      foreach (S3Object obj in listResponse.S3Objects)
+      foreach (var obj in fileNames)
       {
-        if (file?.FileKey == obj.Key)
+        if (file?.FileKey == obj)
         {
           var downloadFileName = file.FileName;
+          var filePath = Path.Combine("/home/godfreyowidi/Downloads/DropboxLike", downloadFileName);
+          
+          // Check if a file with the same name already exists
+          if (System.IO.File.Exists(filePath))
```

Comment:
I believe this is not evaluated client-side, but server-side, therefore it will always evaluate to false. Like with the other comment: did you test this manually to verify that it actually works? I'd love to know what the outcome of the test was if you did.